File: array/range-sum-query-immutable.java
class NumArray {
    private int[] prefix_array;
    public NumArray(int[] nums) {
        prefix_array = new int[nums.length+1];
        prefix_array[0] = 0;
        for(int i = 0; i<nums.length; i++){
            prefix_array[i+1] = prefix_array[i] + nums[i];
        }
    }
    
    public int sumRange(int left, int right) {
        if(left == 0){
            return prefix_array[right+1];
        }else{
            return  prefix_array[right+1] - prefix_array[left] ;
        }

    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */

File: array/range-sum-query-immutable.py
class NumArray:
    
    def __init__(self, nums: List[int]):
        self.nums = nums
        self.prefixnums = []

        total = 0
        for ele in nums:
            total += ele
            self.prefixnums.append(total)

    def sumRange(self, left: int, right: int) -> int:
        if left == 0:
            return self.prefixnums[right]
        else:
            return self.prefixnums[right] - self.prefixnums[left -1]
